soroban-sdk/src/num.rs

```diff
@@ -193,6 +200,17 @@ impl U256 {
         }
     }
 
+    pub fn to_u128(&self) -> Result<u128, ConversionError> {
```

Comment:
I don't feel strongly about this, but I think it might be more common for this type of fallible conversion to return an `Option<>`. For example, the `.checked_add` function in the stdlib on integer types returns `Option<>` when an overflow occurs. https://docs.rs/num/latest/num/trait.CheckedAdd.html

Reply:
I was on the fence about `Result<_, ConversionError>`, and I think your suggestion makes more sense.

I'll get this updated!